python/taichi/lang/__init__.py

```diff
@@ -70,22 +70,55 @@ def reset():
     runtime = get_runtime()
 
 
+class EnvironConfig:
+    def __init__(self, args):
+        self.args = args
+        self.cfg_keys = []
+
+    def add(self, key, default, cast=None):
```

Comment:
nit: I'd suggest not to have `default`. If `key` is not in `self.args`, just use `self.args[key] = getattr(ti.cfg, key)`. The advantage is that we only need to maintain the default values in CompileConfig in this way. WDYT?

Reply:
Then what about:
ti.init(debug=True)
xxx
ti.init()
yyy

will yyy run in debug mode?
this is rhe often case in our test.

Reply:
I don't think the behavior of this example will change with or without my suggest, will it? Of course, if `CompileConfig::debug` in **c++** is `true`, but we assign it a `default=False` in **python**, then things will be different. But IMHO that's what we should avoid, otherwise there will be multiple places defining the default value...

Reply:
Ok, so what about log_level? Will it be back to default after init?